skyportal/handlers/api/spectrum.py

```diff
@@ -411,59 +415,62 @@ def get(self, spectrum_id=None):
         """
 
         if spectrum_id is not None:
-            try:
-                spectrum = Spectrum.get_if_accessible_by(
-                    spectrum_id,
-                    self.current_user,
-                    raise_if_none=True,
-                )
-            except AccessError:
-                return self.error(
-                    f'Could not access spectrum {spectrum_id}.', status=403
+            with self.Session() as session:
+                spectrum = session.scalars(
+                    Spectrum.select(session.user_or_token).where(
+                        Spectrum.id == spectrum_id
+                    )
+                ).first()
+                if spectrum is None:
+                    return self.error(
+                        f'Could not access spectrum {spectrum_id}.', status=403
+                    )
+                comments = (
+                    session.scalars(
+                        CommentOnSpectrum.select(
+                            session.user_or_token,
+                            options=[joinedload(CommentOnSpectrum.groups)],
```

Comment:
Do you have any idea why we join the groups here?

Reply:
Nope.